ggml-cuda.cu

```diff
@@ -213,6 +214,24 @@ static_assert(sizeof(half) == sizeof(ggml_fp16_t), "wrong fp16 size");
         }                                                                               \
     } while (0)
 
+// driver API
+#define CU_CHECK(err)                                                                   \
+    do {                                                                                \
+        CUresult err_ = (err);                                                          \
+        if (err_ != CUDA_SUCCESS) {                                                     \
+            int id;                                                                     \
+            cuDeviceGet(&id, 0);                                                        \
+            const char * err_str;                                                       \
+            cuGetErrorString(err_, &err_str);                                           \
+            fprintf(stderr, "\nCUDA error %d at %s:%d: %s\n", err_, __FILE__, __LINE__, \
+                err_str);                                                               \
+            fprintf(stderr, "%s\n", #err);                                              \
+            fprintf(stderr, "current device: %d\n", id);                                \
+            GGML_ASSERT(!"CUDA error");                                                 \
+        }                                                                               \
+    } while (0)
+
+
```

Comment:
In my test implementation for peer access I was able to simply use `CUDA_CHECK` for the `CUresult`. I don't know whether that works universally though.

Reply:
I didn't work for me, it is a different type and requires a different function to get the error description, `cuGetErrorString`.